pkg/csi/csi_ops.go

```diff
@@ -68,7 +69,11 @@ func (o *validateOperations) ValidatePVC(ctx context.Context, pvcName, namespace
 	if o.kubeCli == nil {
 		return nil, fmt.Errorf("kubeCli not initialized")
 	}
-	return o.kubeCli.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, pvcName, metav1.GetOptions{})
+	pvc, err := o.kubeCli.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, pvcName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return pvc, nil
```

Comment:
Fake clients (in k8s 1.31 client-go sdk) no longer returns nil object in case of error. So we had to add explicit checks

Reply:
hmm shouldn't the caller be checking if error is nil before checking the value anyway? Or does this result in a lot of changes to UTs? Why can't we just return the objects as is?

Reply:
Alternatively, we'll have to refactor tests, instead of checking for nil, we'll have to check for empty structs. My intention was to have consistent values in the return. E.g in case of other types of error in the same function, we are returning nil, but in K8s API error, it returns empty struct. We'll also have to refactor UT accordingly to check for different value in case of different types of errors.